### ISSUE TITLE
Describe (and update) architecture diagrams

### ISSUE THREAD

**Issue by pepopowitz** [camunda/camunda-docs#1829]
Some docs improvements that could help clarify the ownership of different components of a Camunda solution: 

1. Update the architecture diagram on https://docs.camunda.io/docs/components/ to include Connectors. See [this message in Slack](https://camunda.slack.com/archives/C026U8GBNSW/p1678696077658299?thread_ts=1678467495.737109&cid=C026U8GBNSW) for another example diagram that includes Connectors.
2. Describe the architecture diagram on https://docs.camunda.io/docs/components/ in words. 
3. Describe the architecture diagram on https://docs.camunda.io/docs/components/concepts/job-workers/ in words. 

Adding descriptions to the diagrams will benefit in two ways: (1) users without sight will gain recognition of what's in the diagram, and (2) all users will gain clarity on the different components in the diagram.

## Background

I was initially confused about the ownership/hosting of workers in a Camunda project. In reviewing our docs afterward with @christinaausley, we recognized that the 3 items described above might add the clarity I was lacking.

This discussion all started in a [thread in Slack](https://camunda.slack.com/archives/C026U8GBNSW/p1678467495737109).

**Comment by akeller** [comment]
We have accessibility on our roadmap (via https://github.com/camunda/product-hub/issues/345) and I think this fits that epic, but also gives us a clear issue tied to improving these architecture diagrams. We have several issue in the backlog for improving/updating the existing architecture diagrams, but it's been difficult to find an internal DRI for architecture diagrams.

**Comment by akeller** [comment]
This issue feels super stale.
